bundles/org.openhab.binding.playstation/src/main/java/org/openhab/binding/playstation/internal/PS4ArtworkHandler.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.playstation.internal;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigConstants;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link PS4ArtworkHandler} is responsible for fetching and caching
+ * application artwork.
+ *
+ * @author Fredrik Ahlström - Initial contribution
+ */
+@NonNullByDefault
+public class PS4ArtworkHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(PS4ArtworkHandler.class);
+    private final File artworkCacheFolder;
+
+    /** Service id */
+    static final String SERVICE_ID = "playstation";
+
+    /** Service category */
+    static final String SERVICE_CATEGORY = "binding";
+
+    /** Service pid */
+    static final String SERVICE_PID = "org.openhab." + SERVICE_CATEGORY + "." + SERVICE_ID;
+
+    /** Cache folder under $userdata */
+    private static final String CACHE_FOLDER_NAME = "cache";
+
+    PS4ArtworkHandler() {
```

Comment:
Is there some policy/common understanding (OpenHab or other) against package private? I'm still learning Java so I have no strong beliefs either way, but a why goes a long way.

Reply:
Just speaking from personal opinion here as seasoned java developer. But package private is kinda the ugly duckling of access modifiers. Package access isn't used very often because anytime you want to set access between `private` and `public` the goto is `protected` since `protected` includes package protection as well as give subclasses access. Since Java is an object-oriented programming (OOP) language, packages are generally seen as a way to organize code, not as a way to architecturally structure code. Package private is an architectural decision whereas `protected` fits into OOP design.

Personally, I wish `protected` didn't include package private access since it is rarely my intention to give other package classes access to protected fields.

I don't know if my explanation makes any sense, it is hard to put into words.

Reply:
The constructor is now private since the whole class is now static.